core/src/main/scala/com/microsoft/azure/synapse/ml/logging/fabric/CertifiedEventClient.scala

```diff
@@ -3,77 +3,26 @@
 
-import com.microsoft.azure.synapse.ml.fabric.TokenLibrary
-import org.apache.spark.sql.SparkSession
+import com.microsoft.azure.synapse.ml.fabric.FabricClient
+import com.microsoft.azure.synapse.ml.logging.common.PlatformDetails.runningOnFabric
 import spray.json.DefaultJsonProtocol.{StringJsonFormat, _}
 import spray.json._
 
 import java.time.Instant
-import java.util.UUID
-
-import com.microsoft.azure.synapse.ml.logging.common.PlatformDetails.runningOnFabric
-
-object CertifiedEventClient extends RESTUtils {
-
-  private val PbiGlobalServiceEndpoints = Map(
-    "public" -> "https://api.powerbi.com/",
-    "fairfax" -> "https://api.powerbigov.us",
-    "mooncake" -> "https://api.powerbi.cn",
-    "blackforest" -> "https://app.powerbi.de",
-    "msit" -> "https://api.powerbi.com/",
-    "prod" -> "https://api.powerbi.com/",
-    "int3" -> "https://biazure-int-edog-redirect.analysis-df.windows.net/",
-    "dxt" -> "https://powerbistagingapi.analysis.windows.net/",
-    "edog" -> "https://biazure-int-edog-redirect.analysis-df.windows.net/",
-    "dev" -> "https://onebox-redirect.analysis.windows-int.net/",
-    "console" -> "http://localhost:5001/",
-    "daily" -> "https://dailyapi.powerbi.com/")
-
 
+object CertifiedEventClient {
   private lazy val CertifiedEventUri = getCertifiedEventUri
 
-  private def getHeaders: Map[String, String] = {
-    Map(
-      "Authorization" -> s"Bearer ${TokenLibrary.getAccessToken}",
-      "RequestId" -> UUID.randomUUID().toString,
-      "Content-Type" -> "application/json",
-      "x-ms-workload-resource-moniker" -> UUID.randomUUID().toString
-    )
-  }
-
-  private def getCertifiedEventUri: String = {
-    val sc = SparkSession.builder().getOrCreate().sparkContext
-    val workspaceId = sc.hadoopConfiguration.get("trident.artifact.workspace.id")
-    val capacityId = sc.hadoopConfiguration.get("trident.capacity.id")
-    val pbiEnv = sc.getConf.get("spark.trident.pbienv").toLowerCase()
-
-    val clusterDetailUrl = s"${PbiGlobalServiceEndpoints(pbiEnv)}powerbi/globalservice/v201606/clusterDetails"
-    val headers = getHeaders
-
-    val clusterUrl = usageGet(clusterDetailUrl, headers)
-      .asJsObject.fields("clusterUrl").convertTo[String]
-    val tokenUrl: String = s"$clusterUrl/metadata/v201606/generatemwctokenv2"
-
-    val payload =
-      s"""{
-         |"capacityObjectId": "$capacityId",
-         |"workspaceObjectId": "$workspaceId",
-         |"workloadType": "ML"
-         |}""".stripMargin
-
-
-    val host = usagePost(tokenUrl, payload, headers)
-      .asJsObject.fields("TargetUriHost").convertTo[String]
-
-    s"https://$host/webapi/Capacities/$capacityId/workloads/ML/MLAdmin/Automatic/workspaceid/$workspaceId/telemetry"
+  def getCertifiedEventUri: String = {
+    s"${FabricClient.MLWorkloadEndpointAdmin}/telemetry"
   }
 
 
-  private[ml] def logToCertifiedEvents(featureName: String,
+  def logToCertifiedEvents(featureName: String,
```

Comment:
do you need to remove the private[ml] here?

Reply:
This can also be used outside of synapse ml